### ISSUE TITLE
[BUG] Browser install hangs infinitely

### ISSUE THREAD

**Issue by eirikbjornr** [microsoft/playwright#28189]
### System info
- Playwright Version: [v1.39.0]
- Operating System: [macOS 13.6]
- Browser: [Chromium]
- Other info: Intel Core

### Source code

I am unable to install the Chromium browser through `playwright`. This happens:

![Image of a cancelled execution of playwright install](https://github.com/microsoft/playwright/assets/44924723/20dde697-d9d6-4bf7-8779-2cfcc03db66e)

The process hangs infinitely with no information. I have tried running `npx playwright install --with-deps`. Same result.

![Image of a hanging execution of playwright install --with-deps](https://github.com/microsoft/playwright/assets/44924723/5ba49e1c-1bdf-4149-8f76-bbf3fe0bcb2a)

Is there anything I can do to provide more info?

**Comment by mxschmitt** [comment]
Does a manual download work?

```
curl -O -v https://playwright.azureedge.net/builds/chromium/1084/chromium-mac.zip
```

or via the browser.

**Comment by eirikbjornr** [comment]
A manual download does work:

```shell
bjorne01@MC-N365497 ~ % curl -O -v https://playwright.azureedge.net/builds/chromium/1084/chromium-mac.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.213.64:443...
* Connected to playwright.azureedge.net (13.107.213.64) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [329 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [102 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [4641 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [333 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [70 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=WA; L=Redmond; O=Microsoft Corporation; CN=*.azureedge.net
*  start date: Oct  8 04:19:55 2023 GMT
*  expire date: Oct  2 04:19:55 2024 GMT
*  subjectAltName: host "playwright.azureedge.net" matched cert's "*.azureedge.net"
*  issuer: C=US; O=Microsoft Corporation; CN=Microsoft Azure RSA TLS Issuing CA 03
*  SSL certificate verify ok.
* using HTTP/2
* h2 [:method: GET]
* h2 [:scheme: https]
* h2 [:authority: playwright.azureedge.net]
* h2 [:path: /builds/chromium/1084/chromium-mac.zip]
* h2 [user-agent: curl/8.1.2]
* h2 [accept: */*]
* Using Stream ID: 1 (easy handle 0x7faa5300c400)
> GET /builds/chromium/1084/chromium-mac.zip HTTP/2
> Host: playwright.azureedge.net
> User-Agent: curl/8.1.2
> Accept: */*
> 
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0< HTTP/2 200 
< date: Thu, 16 Nov 2023 21:48:16 GMT
< content-type: application/zip
< content-length: 146532972
< last-modified: Sat, 07 Oct 2023 07:46:51 GMT
< etag: 0x8DBC709915E7698
< vary: Origin
< x-ms-request-id: 7373ad48-a01e-0093-2ed6-186219000000
< x-ms-version: 2009-09-19
< x-ms-lease-status: unlocked
< x-ms-blob-type: BlockBlob
< x-azure-ref: 20231116T214816Z-hep0s9138h19tf5nsbwp12m64g000000091g00000000y7nd
< x-cache: TCP_MISS
< accept-ranges: bytes
< 
{ [16054 bytes data]
100  139M  100  139M    0     0  1902k      0  0:01:15  0:01:15 --:--:-- 2526k
* Connection #0 to host playwright.azureedge.net left intact
```

**Comment by eirikbjornr** [comment]
fwiw first I encountered issue https://github.com/microsoft/playwright/issues/23569. I then upgraded from Playwright v1.35.0 to v1.39.0 as suggested in that issue. That's when I got this infinitely hanging download instead of the `Error: end of central directory record signature not found`.

**Comment by mxschmitt** [comment]
I was able to reproduce with this webserver:

```go
package main

import (
	"io"
	"log"
	"net/http"
	"os"
	"strconv"
)

func main() {
    http.HandleFunc("/builds/chromium/1091/chromium-mac-arm64.zip", func(w http.ResponseWriter, r *http.Request) {
		// Open the zip file
        file, err := os.Open("chromium-mac-arm64.zip")
        if err != nil {
            http.Error(w, "File not found.", 404)
            return
        }
        defer file.Close()

        // Set the content type header
        w.Header().Set("Content-Type", "application/zip")
		stat, _ := file.Stat()
        w.Header().Set("Content-Length", strconv.FormatInt(stat.Size(), 10))

        // Limit the data transfer to 50MB
        io.CopyN(w, file, 50<<20) // 50<<20 is 50MB in bytes
    })

    log.Fatal(http.ListenAndServe(":8081", nil))
}
```

Thanks for filing, we'll look into it!

**Comment by kachkaev** [comment]
I faced the same issue and shared some thoughts in https://github.com/microsoft/playwright/issues/22878#issuecomment-1826406962.

**TL;DR:** As a workaround, it might be worth trying a different internet connection before anything else.

**Comment by suqi** [comment]
Try `sudo playwright install`

Works for my macos.

**Comment by gihrig** [comment]
The issue for me, on macOS 15.3, was `~/Library/Caches/ms-playwright` and sub folders were root:staff owned.
I deleted `~/Library/Caches/ms-playwright` and ran `npx playwright install` again and all was well.

**Comment by PeeJeeDR** [comment]
Solution of @gihrig worked for me.

**Comment by sammy-tam** [comment]
@gihrig thanks! this was indeed the issue

**Comment by ScDor** [comment]
`rm -rf ~/.cache/ms-playwright/` did the trick for me

**Comment by jorgecolonconsulting** [comment]
> The issue for me, on macOS 15.3, was `~/Library/Caches/ms-playwright` and sub folders were root:staff owned. I deleted `~/Library/Caches/ms-playwright` and ran `npx playwright install` again and all was well.

Thanks for sharing this! Spent waaaay too much time debugging this.

**Comment by jsjoeio** [comment]
`sudo yarn playwright install` worked for me on macOS 14.7.7

**Comment by cslaviero** [comment]
Just had the same issue on MacOS (26.1). @gihrig solution worked like a charm.

**Comment by AminBlg** [comment]
Arch linux, still same issue, freezes after download. Tried ``rm -rf ~/.cache/ms-playwright``, doesn't fix it. 
Used to work, so you can't say but arch linux is not supported. 
**Solution**, point it to use system browser:
 ``npx -y @playwright/mcp@latest --browser=firefox --executable-path=/bin/firefox``

**Comment by PeterDraex** [comment]
@AminBlg I updated to node 26 today and playwright `1.58.2` installation started failing. `1.60.0` is installing ok.

**Comment by sustacek** [comment]
> [@AminBlg](https://github.com/AminBlg) I updated to node 26 today and playwright `1.58.2` installation started failing. `1.60.0` is installing ok.

Had a similar issues today with 1.58.2 Jenkins CI, running on some company-internal docker image - not sure what it is based off). 1.60.0 seems to be working fine.

**Comment by twk3** [comment]
Also saw today when using the github actions partner arm images. Started happening last night. Cache clear didn't fix. But swapping to x86, or updating to 1.60 did fix it.

**Comment by davepaiva** [comment]
This did not work : `rm -rf ~/.cache/ms-playwright/`
What worked : I updated the package to 1.60.0 

Any idea why it worked after updating the package ?

**Comment by shufps** [comment]
I can confirm, 1.16.0 works 👍

**Comment by descampsk** [comment]
On github actions, the installation seems to be hanging forever since a few day : 
```
No VM guests are running outdated hypervisor (qemu) binaries on this host.
Downloading Chromium 143.0.7499.4 (playwright build v1200) from https://cdn.playwright.dev/dbazure/download/playwright/builds/chromium/1200/chromium-linux.zip
|                                                                                |   0% of 164.7 MiB
|■■■■■■■■                                                                        |  10% of 164.7 MiB
|■■■■■■■■■■■■■■■■                                                                |  20% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■                                                        |  30% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                                |  40% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                        |  50% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                                |  60% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                        |  70% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■                |  80% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■        |  90% of 164.7 MiB
|■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■| 100% of 164.7 MiB
```

1.16.0 still does not work.

```
      - name: Set up Node.js 24
        uses: actions/setup-node@v6
        with:
          node-version: '24'
          cache: 'npm'

      - name: Set up Java 21
        uses: actions/setup-java@v5
        with:
          distribution: 'temurin'
          java-version: '21'

      - name: Install Playwright Chrome
        run: npx playwright install --with-deps chromium
        env:
          DEBIAN_FRONTEND: noninteractive
```

**Comment by descampsk** [comment]
Seems I had to upgrade also "@playwright/test": "1.60.0" 
After both upgrades, it works again

**Comment by jules43** [comment]
Upgrading to 1.60.0 did not fix it for me, however downgrading from Node 24.16 to Node 24.15 explicitly in my deploy.yml does provide a workaround.

**Comment by GollyTicker** [comment]
For me the combination of 1.60.0 and Node 24.16.0 worked. I simply force-reinstalled playwright globally and ran the installation of chromium in a fresh directory and that seemed to fix it for me.

**Comment by colinkershaw** [comment]
Updating Playwright (to 1.60.0) resolved this issue for me as well.
